radix-engine/src/model/component.rs

```diff
@@ -29,19 +36,177 @@ impl Component {
         }
     }
 
-    pub fn get_auth(&self, method_name: &str) -> &MethodAuthorization {
-        match self.auth_rules.get(method_name) {
-            Some(auth_rule) => auth_rule,
-            None => &MethodAuthorization::Public,
+    fn soft_to_hard_resource_list(
+        schema: &Type,
+        list: &SoftResourceOrNonFungibleList,
+        dom: &Value,
+    ) -> HardProofRuleResourceList {
+        match list {
+            SoftResourceOrNonFungibleList::Static(resources) => {
+                let mut hard_resources = Vec::new();
+                for soft_resource in resources {
+                    let resource = Self::soft_to_hard_resource_or_non_fungible(schema, soft_resource, dom);
+                    hard_resources.push(resource);
+                }
+                HardProofRuleResourceList::List(hard_resources)
+            }
+            SoftResourceOrNonFungibleList::Dynamic(schema_path) => {
+                let sbor_path = schema_path.to_sbor_path(schema);
+                if let None = sbor_path {
+                    return HardProofRuleResourceList::SoftResourceListNotFound;
+                }
+
+                match sbor_path.unwrap().get_from_value(dom) {
+                    Some(Value::Vec(type_id, values)) => {
+                        match CustomType::from_id(*type_id).unwrap() {
+                            CustomType::ResourceDefId => HardProofRuleResourceList::List(
+                                values
+                                    .iter()
+                                    .map(|v| {
+                                        if let Value::Custom(_, bytes) = v {
+                                            return ResourceDefId::try_from(bytes.as_slice())
+                                                .unwrap()
+                                                .into();
+                                        }
+                                        panic!("Unexpected type");
```

Comment:
This is one of the assumptions that haven't been well documented. When decoding **ANY** value from a byte array, the decoder won't validate the "bytes" of custom types.

Here, we will have to handle the errors or to reuse `CustomValueValidator` with some refactoring.

Reply:
I think I'm getting you but I'm assuming that we will validate the bytes of custom types in a future PR?

Reply:
When do validate the bytes, when using `ValidateData` validator. 

This reminds me that the `dom: &Value` is the component state, which has already been validated by Radix Engine. We should be good unless we want to get rid of such assumption.